# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLContext.cpp
# Commit: bc1b6e82d67f
# Full Hash: bc1b6e82d67fd6018b946c384efa6be9a5a5af16
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2021-12-10 21:58:52
# Regressor Bug: 1736177
# File Overlap Count: 2
# Description:
#   Bug 1736177 - Part 11. Plumb in OffscreenCanvasDisplayHelper. r=jgilbert
#   
#   This patch integrates OffscreenCanvasDisplayHelper with
#   HTMLCanvasElement, OffscreenCanvas and nsDisplayCanvas to allow
#   asynchronous display of an OffscreenCanvas.
# ==============================================================================

diff -r 7b946576bafa -r bc1b6e82d67f dom/canvas/WebGLContext.cpp
--- a/dom/canvas/WebGLContext.cpp	Fri Dec 10 02:57:55 2021 +0000
+++ b/dom/canvas/WebGLContext.cpp	Fri Dec 10 02:57:55 2021 +0000
@@ -231,6 +231,8 @@
   if (mCanvasElement) {
     SVGObserverUtils::InvalidateDirectRenderingObservers(mCanvasElement);
     mCanvasElement->InvalidateCanvasContent(nullptr);
+  } else if (mOffscreenCanvas) {
+    mOffscreenCanvas->QueueCommitToCompositor();
   }
 }
 